src/App.tsx

```diff
@@ -1,61 +1,109 @@
+import React from 'react';
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { getUserById, todos } from './components/todosStaticList';
+import { TodoList } from './components/TodoList';
+import users from './api/users';
+import { useState } from 'react';
+import { Todo } from './types';
 
 export const App = () => {
+  const [currTodos, setCurrTodos] = useState<Todo[]>(todos);
+  const [titleValue, setTitleValue] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+  const [inputError, setInputError] = useState(false);
+  const [selectError, setSelectError] = useState(false);
+
+  const reset = () => {
+    setTitleValue('');
+    setSelectUser(0);
+    setInputError(false);
+    setSelectError(false);
+  };
+
+  const getlargestId = () => {
+    const newTodos: Todo[] = [...currTodos];
+
+    newTodos.sort((todo1, todo2) => todo1.id - todo2.id);
+
+    return newTodos[newTodos.length - 1].id + 1;
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!titleValue) {
+      setInputError(true);
+    }
+
+    if (!selectUser) {
+      setSelectError(true);
+    }
+
+    if (titleValue && selectUser) {
+      const newTodo: Todo = {
+        id: getlargestId(),
+        title: titleValue,
+        completed: false,
+        userId: selectUser,
+        user: getUserById(selectUser) || null,
+      };
+
+      setCurrTodos(prevTodos => [...prevTodos, newTodo]);
+
+      reset();
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleId">Title: </label>
+          <input
+            id="titleId"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={titleValue}
+            onChange={event => setTitleValue(event.target.value)}
+          />
+          {inputError && !titleValue && (
+            <span className="error">Please enter a title</span>
```

Comment:
It is not necessary, but you can also be output error messages in variables:
```
const EMPTY_TITLE_ERROR = 'Please enter a title';
const EMPTY_USER_ERROR = 'Please choose a user';
```
```suggestion
            <span className="error">{EMPTY_TITLE_ERROR}</span>
```